src/frontend/app/shared/data-services/scm/gitlab-scm.ts

```diff
@@ -97,6 +97,20 @@ export class GitLabSCM implements GitSCM {
     return `https://gitlab.com/${projectName}/compare/${commitSha1}...${commitSha2}`;
   }
 
+  getMatchingRepositories(projectName: string): Observable<string[]> {
+    const prjParts = projectName.split('/');
+    let url = `${gitLabAPIUrl}/projects?search=${projectName}`;
+    if (prjParts.length > 1) {
+      url = `${gitLabAPIUrl}/users/${prjParts[0]}/projects?search=${prjParts[1]}`;
+    }
+    return this.http.get(url).pipe(
```

Comment:
We should use angular's httpClient (https://angular.io/guide/http) as the old Http is deprecated. httpClient removes the need to do response.json()